### PR TITLE
[python] qvector init from lists and arrays

### PR DESCRIPTION
**Pull request by annagrin**
### Description

Towards: https://github.com/NVIDIA/cuda-quantum/issues/1086

Support `qvector` initialization from lists and arrays:

- Allow using lists and arrays of data types to initialize `qvector`
  - As a captured value
  - Passed as a parameter to a kernel
    - NOTE: not supported for `np.ndarray` and `np.array` yet
  - Created inside kernel

- Allow following types of initializers
  - Lists of data types below
  - `np.ndarray` of data types below 
 
- Allow following data types in initializers
  - `float`, `np.float32`, `np.float64`, `complex`, `np.complex128`, `np.complex128`

- Support using `cudaq.complex`, `cudaq.amplitudes`
  - As captures or passed as parameters
  - Inside kernels

- Allow casting of lists of data to the when passed as parameters to the formal type

- Perform automatic conversion of vectors to a vector of simulation data type when needed
  - Issue a performance warning on conversion

### CODE REVIEW

File: python/tests/kernel/test_kernel_state_init.py [hunk 1]
```diff
@@ -0,0 +1,671 @@
...
+    assert '11' in counts
+    assert '00' in counts
+
+
+@skipIfNvidiaNotInstalled
+def test_kernel_float_params_f32():
+    cudaq.reset_target()
+    cudaq.set_target('nvidia')
+
+    f = [1., 0., 0., 1.]
```
Comment (1tnguyen) [inline python/tests/kernel/test_kernel_state_init.py:46]: same as above

File: python/tests/kernel/test_kernel_state_init.py [hunk 2]
```diff
@@ -0,0 +1,671 @@
...
+    assert '11' in counts
+    assert '00' in counts
+
+
+@skipIfNvidiaFP64NotInstalled
+def test_kernel_float_capture_f64():
+    cudaq.reset_target()
+    cudaq.set_target('nvidia-fp64')
+
+    f = [1., 0., 0., 1.]
```
Comment (1tnguyen) [inline python/tests/kernel/test_kernel_state_init.py:63]: same as above

File: python/tests/kernel/test_kernel_state_init.py [hunk 3]
```diff
@@ -0,0 +1,671 @@
...
+    assert '11' in counts
+    assert '00' in counts
+
+
+@skipIfNvidiaNotInstalled
+def test_kernel_float_capture_f32():
+    cudaq.reset_target()
+    cudaq.set_target('nvidia')
+
+    f = [1., 0., 0., 1.]
```
Comment (1tnguyen) [inline python/tests/kernel/test_kernel_state_init.py:80]: same as above

File: python/tests/kernel/test_kernel_state_init.py [hunk 4]
```diff
@@ -0,0 +1,671 @@
...
+    assert '11' in counts
+    assert '00' in counts
+
+
+@skipIfNvidiaFP64NotInstalled
+def test_kernel_float_np_array_from_capture_f64():
+    cudaq.reset_target()
+    cudaq.set_target('nvidia-fp64')
+
+    f = [1., 0., 0., 1.]
```
Comment (1tnguyen) [inline python/tests/kernel/test_kernel_state_init.py:97]: same as above

Comment (annagrin) [reply python/tests/kernel/test_kernel_state_init.py]: Done, thanks!

File: python/tests/kernel/test_kernel_state_init.py (hunk 3 above)
Comment (annagrin) [reply python/tests/kernel/test_kernel_state_init.py]: Done!

File: python/tests/kernel/test_kernel_state_init.py (hunk 2 above)
Comment (annagrin) [reply python/tests/kernel/test_kernel_state_init.py]: Done!

File: python/tests/kernel/test_kernel_state_init.py (hunk 1 above)
Comment (annagrin) [reply python/tests/kernel/test_kernel_state_init.py]: Done!

File: python/tests/kernel/test_kernel_state_init.py [hunk 5]
```diff
@@ -0,0 +1,671 @@
...
+
+
+# float
+@skipIfNvidiaFP64NotInstalled
+def test_kernel_float_params_f64():
+
+    cudaq.reset_target()
+    cudaq.set_target('nvidia-fp64')
+
+    f = [1., 0., 0., 1.]
```
[the remark this answers is not in the document]
Comment (annagrin) [reply python/tests/kernel/test_kernel_state_init.py]: Thanks, done! I updated the PR instead due to many changes that needed to be made.

**Pull request merged**